File: tests_compile/src/test/resources/SimpleCarComponent.java
package com.github.klee0kai.simple.car;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.module.Module;

@Component()
abstract class CarInjectComponent {
    protected abstract CarInjectModule module();
}

@Module
abstract class CarInjectModule {

}

File: tests_compile/src/test/resources/SimpleCarComponent.kt
package com.github.klee0kai.stone.test;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.module.Module;

@Component()
abstract class CarInjectComponent {

    abstract fun module(): CarInjectModule

}

@Module
abstract class CarInjectModule {

}
